python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py

```diff
@@ -2641,6 +2641,10 @@ def aten〇native_layer_norm〡dtype(input_rank_dtype: Tuple[int, int], normaliz
         result_dtype = torch.float64
     return input_dtype, input_dtype, result_dtype
 
+@check_dtype_function([Invocation(TensorOfShape(5, dtype=torch.int64, device="cpu"), 2)])
```

Comment:
Can you do

```python
@check_dtype_function(_check_tensors_with_the_same_dtype(num_of_tensors=1, num_classes=2))
```

This performs a much more thorough check

Reply:
I think it's no need to add this check, torch defines that the input/output tensor must be int64 dtype. See:
https://pytorch.org/docs/stable/generated/torch.nn.functional.one_hot.html

I will add `assert` check in dtype function.

Reply:
Sometimes docs can become out of date and inconsistent with actual behavior of functions. The only way to know that things work properly is by actually running them. This also makes it easy to know when API changes happen in PyTorch that affect dtype calculations. I would prefer to use the comprehensive check.

Reply:
Done, could you please reivew this PR again? Thanks!